php/blocks/migration/class-post-content.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Post_Content.
+ *
+ * @package   Block_Lab
+ * @copyright Copyright(c) 2020, Block Lab
+ * @license http://opensource.org/licenses/GPL-2.0 GNU General Public License, version 2 (GPL-2.0)
+ */
+
+namespace Block_Lab\Blocks\Migration;
+
+use WP_Error;
+
+/**
+ * Class Post_Content
+ */
+class Post_Content {
+
+	/**
+	 * The previous namespace of the block.
+	 *
+	 * @var string
+	 */
+	private $previous_block_namespace;
+
+	/**
+	 * The new namespace of the block.
+	 *
+	 * @var string
+	 */
+	private $new_block_namespace;
+
+	/**
+	 * Post_Content constructor.
+	 *
+	 * @param string $previous_block_namespace Previous namespace of the blocks.
+	 * @param string $new_block_namespace      New namespace of the blocks.
+	 */
+	public function __construct( $previous_block_namespace, $new_block_namespace ) {
+		$this->previous_block_namespace = $previous_block_namespace;
+		$this->new_block_namespace      = $new_block_namespace;
+	}
+
+	/**
+	 * Migrates all of the block namespaces in all of the posts that have Block Lab blocks.
+	 *
+	 * @return array The results of migration, either int or WP_Error.
+	 */
+	public function migrate_all() {
+		$posts   = $this->query_for_posts();
+		$results = [];
+
+		while ( $posts ) {
+			foreach ( $posts as $post ) {
+				if ( isset( $post->ID ) ) {
+					$results[] = $this->migrate_single( $post->ID );
+				}
+			}
+
+			$posts = $this->query_for_posts();
+		}
+
+		return $results;
```

Comment:
That's a good catch, I'll think about that. It could definitely create an infinite loop if there are 10 failures.

Reply:
Really good point, I think that is a case we should handle.

What do you think about https://github.com/getblocklab/block-lab/pull/606/commits/f6aada67ce83125c46285ed04cce24375ed3389c?

It returns from `migrate_all()` if there are more than 20 `WP_Error`s in migration.

It'll prevent an infinite loop. 

If there was an `WP_Error`, something might have gone very wrong. Like `wp_insert_post()` [tried to insert an empty post](https://github.com/WordPress/wordpress-develop/blob/5e2ef3d93739610e97dd20f5f254a355a4eb5bd3/src/wp-includes/post.php#L3697). In that case, it's probably better of exiting.

It might be possible to keep track of failed post IDs, and exclude them from the SQL query, maybe with `NOT IN`. But that might not be very efficient.

Reply:
Yeah, this is all pretty edge case. I'm happy with the 20 errors limit. 🚢

Reply:
Nice, thanks!